sandbox/gamedev/unity/samples/azure-facebook-auth-unity.md

```diff
@@ -0,0 +1,646 @@
+
+  ![create function](media/fbauth_create-function1.png)
+
+4. Select **HTTP trigger**.
+
+5. Choose **C#** from the **Language:** dropdown.
+
+6. Name the function `Insert`.
+
+7. Select **Anonymous** from the **Authorization level** dropdown.
+
+8. Click the **Create** button.
+
+  ![new function](media/fbauth_new-function.png)
+
+#### Add the Azure Mobile Client and Newtonsoft.Json NuGet packages to the Insert function
+This example uses the Azure Mobile Client SDK to simplify authentication and data operations.
+
+1. Once the function has been created, select **View files** from the right side panel. You may need to extend the window or scroll right to find it.
+
+2. Click **+ Add**.
+
+3. Enter **project.json** as the new file's name and press Enter on the keyboard to create the file.
+
+  ![add project file](media/fbauth_add-project-file.png)
+
+4. Once the new project.json file is created, select it in the **View files** list.
+
+5. Paste the following lines into the empty project.json file:
+  ```
+  {
+  "frameworks": {
+    "net46":{
+      "dependencies": {
+        "Microsoft.Azure.Mobile.Client": "4.0.2"
+      }
+    }
+   }
+}
+```
+6. Click **Save**. If you observe the log, it should detail that the Microsoft.Azure.Mobile.Client and Newtonsoft.Json packages have been installed and compilation has succeeded.
+
+  ![packages added](media/fbauth_packages-added.png)
+
+#### Add the Insert function body
+
+The Insert function receives Json parameters in an HTTP POST request sent from the Unity client.
+
+The parameters include:
+
+  * The name of the table to be modified.
+
+  * The access token for Facebook authentication (the Facebook Unity SDK gets this key after the user logs in).
+
+  * The data object to add to the table.
+
+This insert function deserializes the Json parameters, and then logs in and inserts the data into the appropriate table using the Azure Mobile Apps SDK.
+
+1. Click on **run.csx** in the **View files** list to modify the function body.
+
+2. Clear all of the initial code in the function and replace it with the following, then click **Save**:
+
+```csharp
+using System.Net;
+using Microsoft.WindowsAzure.MobileServices;
+using Newtonsoft.Json.Linq;
+
+public static async Task<HttpResponseMessage> Run(HttpRequestMessage req, TraceWriter log)
+{
+    log.Info("C# HTTP trigger function processed a request.");
+    MobileServiceClient client = new MobileServiceClient("https://REPLACE-WITH-YOUR-MOBILEAPP-URL");
+    dynamic data = await req.Content.ReadAsStringAsync();
+
+    // Server expects a json arrary with the format:
+    // [{"access_token":"value"},{"tableName":"value"},{instanceJson}]
+    JArray arrayJson = JArray.Parse(data);
+    dynamic authToken = arrayJson[0];
+    var tableName = arrayJson[1].Value<string>("tableName");
+    dynamic objectToInsert = arrayJson[2];
+
+    // Try to log in. Return Unauthorized response upon failure.
+    MobileServiceUser user = null;
+    while (user == null)
```

Comment:
I used this code to follow the example in the Azure Mobile Apps client SDK documentation, as found here: https://docs.microsoft.com/en-us/azure/app-service-mobile/app-service-mobile-dotnet-how-to-use-client-library#client-facebook

If you think omitting the while loop is preferable, I am happy to change, but I wonder if we should consider raising the issue on the other document for consistency?

Reply:
Ah, OK, now I see. I don't know, in my eyes, a simple "if" would suffice, but maybe the original author intended for something else (???). @BrianPeek any thoughts?

Reply:
On a second reading of the "while" code (and seeing that the original author in the article you mentioned has the same "while" 4 times in total in the documentation), I suppose that they just do it to protect from whatever error Login method might bump into. I propose to leave the "while" as is, just have a link that refers to the original article as the source for the "while" way. @BrianPeek do you agree?

Reply:
I dunno, I think that while loop is super weird.  If there was an error that didn't get thrown as an exception, the app would hang forever in that loop.  I think an "if" statement is better...I can't see why one would write it with the "while" loop.

Reply:
Well, let's add @conceptdev who is listed as author on the original doc.  Craig, any reason why the code referenced at https://docs.microsoft.com/en-us/azure/app-service-mobile/app-service-mobile-dotnet-how-to-use-client-library#client-facebook would use a while() loop instead of an if() statement or some other construct?  Seems like a place where an app could loop forever in a wonky circumstance, but perhaps I'm missing something...thanks!